docs/reference/bpmn-processes/error-events/error-events.md

```diff
@@ -3,49 +3,68 @@ id: error-events
+![process](assets/error-throw-events.png)
 
-In case the process instance is created via call activity, the error can also be caught in the calling parent process instance.
+Alternatively, you can inform Zeebe that a business error has occurred using a **client command**.
+This throw error client command can only be used while processing a job.
+In addition to throwing the error, this also disables the job and stops it from being activated or completed by other job workers.
+See the [gRPC command](/reference/grpc.md#throwerror-rpc) for details.
 
-## Throwing the error
+## Catching the error
 
-An error can be thrown from a **client command** while processing a job. See the [gRPC command](/reference/grpc.md#throwerror-rpc) for details.
+A thrown error can be caught by an error catch event.
+Specifically, using an error **boundary event** or an error **event subprocess**.
 
-Alternatively, an error can also be thrown inside a process using an error **end event**.
+Starting at the scope where the error was thrown, the error code is matched against the attached error boundary events and error event sub processes at that level.
+An error is caught by the first event in the scope hierarchy that matches the error code.
+At each scope, the error is either caught, or it is propagated to the parent scope.
+In case the process instance is created via call activity, the error can also be caught in the calling parent process instance.
 
-![process](assets/error-throw-events.png)
+Error boundary events and error event subprocesses must be interrupting.
+This means the process instance will not continue along the regular path when the error was caught.
+Instead, it continues where the error is caught.
+If the error is thrown for a job, then the associated task gets terminated first.
+To continue the execution, the error boundary event or error event subprocess that caught the error gets activated.
 
 ## Unhandled errors
 
-When an error is triggered then it should be handled in the process. If it is not handled (e.g. unexpected error code) then an **incident** is raised to indicate the failure. The incident is attached to the corresponding service task of the processed job or the error end event.
+When an error is thrown then it should be caught.
+If it is not caught, then an **incident** (i.e. `Unhandled error event`) is raised to indicate the failure.
+The incident is attached to the corresponding element where the error was thrown, i.e. the task of the processed job or the error end event.
+
+When you resolve the incident attached to a task, it will ignore the error, re-enable the job and allow it to be activated and completed by a job worker once again.
```

Comment:
(optional) We could improve the wording a bit. The part "it will ignore the error" could be misleading.

  ```suggestion
Resolving the incident attached to a task will re-enable the job and allow it to be activated and completed by a job worker once again.
```

Reply:
How do you feel about `it will swallow the error`? Point is that the error will not be caught anymore. It was thrown, but we just ignore it.

Reply:
I don't know :sweat_smile: I find this part more confusing than helpful. But it could be just me. Feel free to merge it as it is :+1: